forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java

```diff
@@ -0,0 +1,182 @@
+
+                if(username != null) {
+                    //creates the username bundle
+                    Bundle bundle = new Bundle();
+                    bundle.putString(KEY_USERNAME, username);
+
+                    //creates an instance of the fragment to be launched
+                    ShowPosts showPostsFragment = new ShowPosts();
+                    //passes the bundle to the fragment as an argument
+                    showPostsFragment.setArguments(bundle);
+
+                    //performs the fragment transaction
+                    getActivity()
+                            .getSupportFragmentManager()
+                            .beginTransaction()
+                            .replace(
+                                    R.id.fragmentContainer,
+                                    showPostsFragment
+                            )
+                            .commit();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_USERNAME);
+                }
+            }
+
+            /**
+             * The method of the listener is triggered when the database response to
+             * the login request is negative and the view model requests to show the
+             * user an error message, that is performed through the use of a toast
+             *
+             * @param message The error message about what was wrong with the login request
+             * @author Alberto Valente (g2), Taulant Bullaku (g2)
+             */
+            @Override
+            public void onLoginFailure(String message) {
+
+                if (message != null) {
+                    Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_ERROR_MESSAGE);
+                }
+            }
+        });
+
+        //When the login button is clicked a request is sent through the view model
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                //The username and password inserted are gathered from the EditText objects
+                userName = usernameEditText.getText().toString();
+                userPwd = pwdEditText.getText().toString();
+
+                if (!userName.equals("") && !userPwd.equals(""))
+                {
+                    //Then the credentials are handed to the viewmodel method to be checked
+                    viewModel.loginToForum(getContext(), userName, userPwd);
+                }
+                else
+                {
+                    Toast.makeText(getContext(), "Please enter username and password", Toast.LENGTH_LONG).show();
+                }
+
+            }
+        });
+
+    }
+
+    public interface OnFragmentInteractionListener
+    {
+        // TODO: Update argument type and name
+        void onFragmentInteraction(Uri uri);
+    }
```

Comment:
I dont understand the meaning of this interface. Is that used somewhere? Does it needs to be completed? Otherwise i suggest u to delete it or at least complete the TODO.